packages/cdktf-cli/bin/cmds/helper/synth-stack.ts

```diff
@@ -26,7 +26,8 @@ interface ManifestJson {
 export class SynthStack {
   public static async synth(
     command: string,
-    outdir: string
+    outdir: string,
+    graceful = false // will not exit the process but rethrow the error instead
```

Comment:
Nitpick: We need graceful due to the `process.exit` call as far as I understand. We want to get rid of the process.exit midterm since they interfere with telemetry and possibly other things. Why not switch to throwing errors here in general?

Reply:
Because I did not know what kind of repercussion that might cause for the existing commands using it 😅